### PR TITLE
Make canvas zoom permanent

### LINKED ISSUE #6913: Add tests for canvas zoom
## Feature Description

Add relevant integration tests for canvas zoom.

---

_Do not alter or remove anything below. The following sections will be managed by moderators only._

### LINKED ISSUE #6914: Remove zoom feature flag
## Feature Description

Remove zoom feature flag and make sure entire feature passes QA.

---

_Do not alter or remove anything below. The following sections will be managed by moderators only._

### PR DESCRIPTION
**Pull request by barklund**
## Context

This removes the canvas zoom feature flag and makes the feature permanent.

It also adds tests for the various zoom settings.

### QA

Please do a full test of everything relevant for the zoom feature and mention any bugs or weirdness not already reported.

Known bugs include:

* New elements are added in weird places and sizes - sometimes even outside the current viewport
* If editing a text element or media element, the canvas cannot scroll - scroll only works once you exit edit-mode
* Pinch to zoom does not work

### UAT

- [x] UAT should use the same steps as above.

## Checklist

- [x] This PR addresses an existing issue and I have linked this PR to it in ZenHub
- [x] I have tested this code to the best of my abilities
- [x] I have verified accessibility to the best of my abilities ([docs](https://github.com/google/web-stories-wp/blob/main/docs/accessibility-testiing.md))
- [x] I have verified i18n and l10n (translation, right-to-left layout) to the best of my abilities
- [x] This PR contains automated tests (unit, integration, and/or e2e) to verify the code works as intended ([docs](https://github.com/google/web-stories-wp/tree/main/docs#testing))
- [x] I have added documentation where necessary
- [x] I have added a matching `Type: XYZ` label to the PR

---

Fixes #6913
Fixes #6914

### CODE REVIEW

File: assets/src/edit-story/app/layout/useZoomSetting.js
Comment (miina) [inline assets/src/edit-story/app/layout/useZoomSetting.js]: ^ There seems to be still `hasCanvasZoom` in use, e.g. on the above line and also in the `LayoutProvider` when looking at the code.

Should we remove it fully since the feature flag was removed now?

**Review by miina — APPROVED** [review]
Awesome to have the tests now, too!

There seem to be some leftover uses of `hasCanvasZoom`. Since it doesn't have any influence, approving this not to block the PR from being QA-ed already, however, let's remove all the references from the code before merging.

File: assets/src/edit-story/app/layout/useZoomSetting.js
Comment (barklund) [reply assets/src/edit-story/app/layout/useZoomSetting.js]: Good catch, fixed in 959ec53b4.

**Comment by swissspidy** [comment]
> This removes the canvas zoom feature flag and makes the feature permanent.

As per our [documentation](https://github.com/google/web-stories-wp/blob/46cda7eddba0a8db82285842ffbf09786cc92e88/docs/feature-flags.md#enabling-feature-flags-by-default), feature flags should first be enabled by default in `Experiments.php` so that we could easily revert the change in case of issues.

The complete removal & code cleanup can be done in a separate PR sometime later.

Something to remember for next time :-)

> It also adds tests for the various zoom settings.

Is there a reason why these weren't added earlier?

In case it wasn't known, Karma tests can totally be added for features behind a flag.

To enable the flag for the test, use `fixture.setFlags({hasCanvasZoom: true});`

**Review by swissspidy — APPROVED** [review]

**Pull request merged**
